hbase-http/src/main/java/org/apache/hadoop/hbase/util/JSONBean.java

```diff
@@ -237,6 +267,22 @@ private static void writeAttribute(JsonWriter writer, MBeanServer mBeanServer, O
     if (attName.indexOf("=") >= 0 || attName.indexOf(":") >= 0 || attName.indexOf(" ") >= 0) {
       return;
     }
+
+    if (pattern != null) {
+      boolean matchFound = false;
+      for (String patt : pattern) {
+        Pattern compile = Pattern.compile(patt);
```

Comment:
We are compiling this pattern for each and every MBean attribute.  This will be costly. we could have compiled the attribute and kept that in an Array at 1st step itself (instead of keeping the regex patterns in String[])

Reply:
Ya ok . I did not do it thinking we are not doing this thing much often. May be once a while. I can change it though.